### PR TITLE
[MINOR][DOCS]Document subquery support for the `table` parameter

### PR DESCRIPTION
**Pull request by wdick**
Hi, yesterday I spend some time figuring out, how to read only the data I really need from a JDBC connection. It took me way too long.
So, I decided to document the solution.

Hopefully it will save others like me, which are new to spark, to save some time.

### CODE REVIEW

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 1]
```diff
@@ -235,6 +235,19 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
    * Construct a `DataFrame` representing the database table accessible via JDBC URL
    * url named table and connection properties.
    *
+   * The `table` parameter can be based on a derived table, which is generated within the scope of
+   * a query FROM clause. For example, you could use this subquery as `table:
+   * {{{
+   *    "(SELECT * FROM customers WHERE ...) AS tmp"
+   * }}}
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: This duplicates `docs/sql-programming-guide.md`. I don't think we have to add these.

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala [hunk 2]
```diff
@@ -235,6 +235,19 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
...
+   *
+   * @param url JDBC database url of the form `jdbc:subprotocol:subname`.
+   * @param table Name of the table in the external database.
+   * @param properties JDBC database connection arguments, a list of arbitrary string
+   *                             tag/value. Normally at least a "user" and "password" property
+   *                             should be included. "fetchsize" can be used to control the
+   *                             number of rows per fetch and "queryTimeout" can be used to wait
+   *                             for a Statement object to execute to the given number of seconds.
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: nit: let's make the indentation consistent

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala (hunk 1 above)
Comment (wdick) [reply sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: Perfect! This is a part of the documentation, that I seam to have missed.

File: sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala (hunk 2 above)
Comment (wdick) [reply sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: I will close the PR, so no need to fix it.
What does 'nit' mean?

Comment (HyukjinKwon) [reply sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala]: I meant not a big deal by nit. Yea, I think we can close

**Pull request closed without merging**